src/core/flow.js

```diff
@@ -418,12 +437,15 @@ export function FlowWithPhases({
   }
 
   function processMove(state, action, dispatch) {
-    // Update currentPlayerMoves.
-    const currentPlayerMoves = state.ctx.currentPlayerMoves + 1;
-    state = { ...state, ctx: { ...state.ctx, currentPlayerMoves } };
-
     const conf = phaseMap[state.ctx.phase];
 
+    const currentPlayerMoves = state.ctx.currentPlayerMoves + 1;
+    const actionPlayers = conf.turnOrder.actionPlayers(G, state.ctx);
```

Comment:
I would imagine that the `actionPlayer` would change in response to some event (someone played the Militia card in Dominion, for example). Rather than have that logic sit inside a `TurnOrder`, I think it's more natural inside a move:

```
playMilitia(G, ctx) {
   Events.setActionPlayers(...);
}
```

I'm not asking you to implement https://github.com/google/boardgame.io/issues/142 in this PR (you can if you like), but let's make sure that this PR is compatible with that approach.

Reply:
Actually, this might also suggest not retiring 'any' so that it's easy for the user. At the very least, we should convert it to an enum (constant) rather than a hardcoded string.